### DISCUSSION TITLE
ENV vars are not expanded, causing DRY violations

### DISCUSSION THREAD

**Discussion by eduncan911** [community/community#25850] [Actions]
<p><strong>Problem:</strong></p>
<p>The Jobs -&gt; Action -&gt; ENV specification does not expand existing ENVs.</p>
<pre><code>name: Go
on: [push]
jobs:

  build:
    name: Build
    runs-on: ubuntu-latest
    strategy:
      fail-fast: true
    env:
      CODE: ~/go/src/github.com/${GITHUB_REPOSITORY}

    - name: Print Code
      run: echo ${CODE}
</code></pre>
<p><strong>Expected:</strong></p>
<p>    /home/runner/go/src/github.com/eduncan911/github-actions</p>
<p><strong>Actual Results:</strong></p>
<p>    ~/go/src/github.com/${GITHUB_REPOSITORY}</p>
<p>It does not expand the <code>~</code> tilda, nor the <code>${GITHUB_REPOSITORY}</code> environment variables.</p>

**Comment by ethomson** [comment, marked as answer]
<p>No, that’s not expanded since the YAML is not evaluated in the context of the bourne shell.  You can use the YAML variable expansion, though:</p>
<pre><code>name: Go
on: [push]
jobs:

  build:
    name: Build
    runs-on: ubuntu-latest
    strategy:
      fail-fast: true
    env:
      CODE: ${{ github.workspace }}/go/src/github.com/${{ github.repository }}

    - name: Print Code
      run: echo ${CODE}
</code></pre>

**Comment by eduncan911** [comment]
<p>Nice! I’ll give that a shot and will report back.</p>
